### PR TITLE
proton-caller: init at 2.2.3

### PR DESCRIPTION
**Pull request by ghost**
###### Motivation for this change
Added this package for being able to test programs using Proton. This can be useful for running some non-Steam games and other Windows programs.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 121471` at 3b5b8822 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>proton-caller</li>
  </ul>
</details>
<details>
  <summary>2 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md">unclear-gpl</a>

`gpl3` is a deprecated license, please check if project uses `gpl3Plus` or `gpl3Only` and change `meta.license` accordingly.

Near pkgs/misc/emulators/proton-caller/default.nix:29:5:
```
   |
29 |     license = licenses.gpl3;
   |     ^
```
</li>
    <li>warning: unused-argument

Unused argument: `stdenv`.
Near pkgs/misc/emulators/proton-caller/default.nix:1:8:
```
  |
1 | { lib, stdenv, fetchFromGitHub, rustPlatform, installShellFiles }:
  |        ^
```
</li>
  </ul>
</details>

File: pkgs/misc/emulators/proton-caller/default.nix [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+  nativeBuildInputs = [ installShellFiles ];
+
+  outputs = [ "out" "man" ];
+
+  postInstall = ''
+    installManPage manual/proton-call.6
+  '';
+
+  doCheck = true;
```
Comment (legendofmiracles) [inline pkgs/misc/emulators/proton-caller/default.nix:24]:
```suggestion
```

**Review by legendofmiracles — APPROVED** [review]
lgtm

**Comment by legendofmiracles** [comment]
I didn't see that yesterday, can you squash all package related commits into one?

**Comment by legendofmiracles** [comment]
That doesn't look like one commit to me 🤔

**Comment by Kho-Dialga** [comment]
> That doesn't look like one commit to me

How can I squash multiple commits into one? I'm kind of new to git.

**Comment by legendofmiracles** [comment]
sorry for not responding @Kho-Dialga 
you have to `git rebase -i HEAD~7` and then `fixup` all the commits into one. You can find plenty of tutorials online

File: pkgs/misc/emulators/proton-caller/default.nix [hunk 2]
```diff
@@ -0,0 +1,31 @@
+{ lib, fetchFromGitHub, rustPlatform, installShellFiles }:
+
+rustPlatform.buildRustPackage rec {
+  pname = "proton-caller";
+  version = "2.2.3";
+
+  src = fetchFromGitHub {
+    owner = "caverym";
+    repo = pname;
+    rev = "${version}";
```
Comment (SuperSandro2000) [inline pkgs/misc/emulators/proton-caller/default.nix:10]:
```suggestion
    rev = version;
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by Kho-Dialga** [comment]
Turns out that I accidentally merged the base nixpkgs repos with my fork with proton-caller and when I try to squash all my commits, instead of getting just the ones that I did, I get prompted to squash a lot more commits. What should I do now?

**Comment by legendofmiracles** [comment]
You can `git reset --hard 46c3d03e0de4c11a658772b81b49a30edfd92490`, that will move HEAD to the last commit of this PR, and drop all the newer ones

File: pkgs/misc/emulators/proton-caller/default.nix [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  postInstall = ''
+    installManPage manual/proton-call.6
+  '';
+
+  meta = with lib; {
+    description = "Run Windows programs with Proton";
+    changelog = "https://github.com/caverym/proton-caller/releases/tag/${version}";
+    homepage = "https://github.com/caverym/proton-caller";
+    license = licenses.gpl3Only;
+    maintainers = with maintainers; [ "kho-dialga" ];
```
Comment (SuperSandro2000) [inline pkgs/misc/emulators/proton-caller/default.nix:29]:
```suggestion
    maintainers = with maintainers; [ kho-dialga ];
```

Please add yourself as a maintainer by adding an alphabetically sorted entry to `maintainers/maintainer-list.nix` and commit this change into a separate commit with the message ``maintainers: add NAME``.

**Comment by legendofmiracles** [comment]
Now the maintainers commit also adds the package...

File: maintainers/maintainer-list.nix [hunk 4]
```diff
@@ -5164,6 +5164,12 @@
     githubId = 546087;
     name = "Kristoffer K. Føllesdal";
   };
+  kho-dialga = {
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:5167]: Please put this into a separate commit with the message ``maintainers: add kho-dialga`` and sort it alphabetically.
